src/UnitTests/IMappingExpression/IncludeMembers.cs

```diff
@@ -1400,4 +1400,46 @@ public class CreateCustomerDto : CustomerDtoBase
         [Fact]
         public void Should_override_IncludeMembers() => Mapper.Map<CreateCustomerDto>(new NewCustomer { Postcode = "Postcode", Address = new Address() }).Postcode.ShouldBe("Postcode");
     }
+    public class IncludeMembersWithValueTypeValidation : AutoMapperSpecBase
+    {
+        class Source
+        {
+            public InnerSource InnerSource { get; set; }
+        }
+        struct InnerSource
+        {
+            public string Name { get; set; }
+        }
+        class Destination
+        {
+            public string Name { get; set; }
+        }
+        protected override MapperConfiguration Configuration => new MapperConfiguration(cfg =>
+        {
+            cfg.CreateMap<Source, Destination>().IncludeMembers(s => s.InnerSource);
+            cfg.CreateMap<InnerSource, Destination>(MemberList.None);
+        });
+    }
+    public class IncludeMembersFromUnconstrainedGenericValidation : AutoMapperSpecBase
+    {
+        class Source<T>
+        {
+            public T InnerSource { get; set; }
+        }
+        class InnerSource
+        {
+            public string Name { get; set; }
+        }
+        class Destination
+        {
+            public string Name { get; set; }
+        }
+        protected override MapperConfiguration Configuration => new MapperConfiguration(cfg =>
+        {
+            var map = cfg.CreateMap<Source<InnerSource>, Destination>();
+            IncludeMembersFromUnconstrainedGeneric(map);
+            cfg.CreateMap<InnerSource, Destination>(MemberList.None);
+        });
+        private static void IncludeMembersFromUnconstrainedGeneric<T>(IMappingExpression<Source<T>, Destination> map) => map.IncludeMembers(src => src.InnerSource);
```

Comment:
I don't see the need for this function.

Reply:
Without this function, the compiler can see that the type parameter (`InnerSource`) is a reference type and does not include a casting expression in the expression tree.

Reply:
Just remove this test then. It's weird and not a use case we encourage.

Reply:
I was also very hesitant to include it, so ok I'll remove it.